projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -133,4 +183,33 @@ export class CartesianLegend {
       })
     );
   }
+
+  private resetToDefault(): void {
+    this.activeSeries = [...this.initialSeries];
+    this.isDefault = true;
```

Comment:
this seems duplicative of a length comparison of active and initial series, can we calculate it on the fly so we don't duplicate information?

Reply:
Actually, this is not. This represents whether we're on the initial/default/reset stage (Where color of the text would be $gray-9).
In a scenario where suppose two legend entries are there, X and Y. If I select A,  then X becomes active ($blue-4) and Y becomes inactive ($gray-5), then if I click on the Y again, then Y also becomes active.

We can implement this in two ways-
1. Whenever the all the entires become active then put them to default state with $gray-9
2. Or we can have them the active state ($blue-4) with the indication of that they have made some changes, anyways they have one click reset button to go to default.

I like the second one more and implemented the same.

Reply:
Yeah that makes sense. I go back and forth on it, it seems weird to have all series selected and it be possible to be in two different states, but it's even weirder to have the selection action change as a user selects - that's what you were describing right? Where if I select A, then select B, then select B again it should deselect B. Selecting a series means "exclusively this" from default, but it means "toggle this" after that.

Reply:
Given that, maybe a slight wording change would help - something to convey we're in a selection mode, even though we may have the default series.

Reply:
> Where if I select A, then select B, then select B again it should deselect B

Correct

Reply:
> something to convey we're in a selection mode, even though we may have the default series.
We have that color change as well as that reset button, I think that can be good enough indication.

Reply:
I meant to the reader of code. Default here refers to a selection mode, rather than which series are selected - the latter would be my assumption as a reader.

Reply:
changed this!!